ocp_resources/utils.py

```diff
@@ -17,15 +17,32 @@ def __str__(self):
 
 class TimeoutSampler:
     """
-    Samples the function output.
-
-    This is a generator object that at first yields the output of function
-    `func`. After the yield, it either raises instance of `TimeoutExpiredError` or
-    sleeps `sleep` seconds.
-
-    Yielding the output allows you to handle every value as you wish.
-
-    Feel free to set the instance variables.
+    Yields a given function's return value.
+
+    Before each following iteration (the next yield) over the TimeSampler object, the generator sleeps for a given
+    sleep interval (seconds). This cycle will be repeated as the caller continues iterating the TimeoutSampler.
+    When the defined timeout expires, a TimeoutExpiredError exception is thrown.
+
+    Args:
+        wait_timeout (int): total timeout (seconds)
+        sleep (int): interval time to wait between each iteration (seconds)
+        func (function): the function to be invoked upon each yield
+        exceptions (Exception or tuple, default Exception): expected exception(s) to ignore when running the generator
+            code
+        exceptions_msg (str, default None): expected exception(s) message; if an exception is caught, but exceptions_msg
+            could not be found in the exception's expression (str), the exception (that was caught) is re-raised after
+            recording the error.
+        print_log (bool, default True): toggle to record timer-related info logs.
+        func_kwargs (dict): function **kwargs
+
+    Yields:
+        the return value of the function func
+
+    Raises:
+        TimeoutExpiredError (Exception): when reaching the wait_timeout.
+        <Exception>: any exception which is not self.exception.
```

Comment:
Can we clarify this slightly? maybe: `any exception which was not purposely ignored`

Reply:
done